corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
+    def get(cls, doc_id):
```

Comment:
Mostly agree with these sentiments, also saw that it conflicts with how `CachedCouchDocumentMixin` does things, which isn't actually so bad, as long as nothing goes wrong :p. The way it handles this is by caching `get`s, but then also invalidating that cache on `save`, `save_docs`, and `delete`, making it, yes, cache implicitly with no way to override the caching, but also makes this caching pretty airtight. One way to break the way the  abstraction works is to call `MyDoc.get_db().save_doc(doc)` or `MyDoc.get_db().bulk_save(docs)`, but as long as you do single saves with `doc.save()` and bulk saves with `MyDoc.save_docs(docs)`, this is pretty airtight.